android/src/main/java/name/avioli/unilinks/UniLinksPlugin.java

```diff
@@ -3,111 +3,139 @@
-    if (call.method.equals("getInitialLink")) {
-      result.success(initialLink);
-      // } else if (call.method.equals("getLatestLink")) {
-      //   result.success(latestLink);
-    } else {
-      result.notImplemented();
+
+    /** Plugin registration. */
+    public static void registerWith(PluginRegistry.Registrar registrar) {
+        // Detect if we've been launched in background
+        if (registrar.activity() == null) {
+            return;
+        }
+
+        final UniLinksPlugin instance = new UniLinksPlugin();
+        instance.context = registrar.context();
+        register(registrar.messenger(), instance);
+
+        instance.handleIntent(registrar.context(), registrar.activity().getIntent());
+        registrar.addNewIntentListener(instance);
     }
-  }
-
-  @Override
-  public void onListen(Object arguments, EventSink events) {
-    changeReceiver = createChangeReceiver(events);
-    // registrar.activity().registerReceiver(
-    // changeReceiver, new IntentFilter(Intent.ACTION_VIEW));
-  }
-
-  @Override
-  public void onCancel(Object arguments) {
-    // registrar.activity().unregisterReceiver(changeReceiver);
-    changeReceiver = null;
-  }
-
-  @Override
-  public boolean onNewIntent(Intent intent) {
-    handleIntent(registrar.context(), intent, false);
-    return false;
-  }
-
-  private BroadcastReceiver createChangeReceiver(final EventSink events) {
-    return new BroadcastReceiver() {
-      @Override
-      public void onReceive(Context context, Intent intent) {
-        // NOTE: assuming intent.getAction() is Intent.ACTION_VIEW
-
-        // Log.v("uni_links", String.format("received action: %s", intent.getAction()));
 
-        String dataString = intent.getDataString();
+    @Override
+    public void onDetachedFromEngine(FlutterPluginBinding flutterPluginBinding) {
+
+    }
 
-        if (dataString == null) {
-          events.error("UNAVAILABLE", "Link unavailable", null);
+    @Override
+    public void onListen(Object o, EventChannel.EventSink eventSink) {
+        changeReceiver = createChangeReceiver(eventSink);
+    }
+
+    @Override
+    public void onCancel(Object o) {
+        changeReceiver = null;
+    }
+
+    @Override
+    public void onMethodCall(MethodCall call, MethodChannel.Result result) {
+        if (call.method.equals("getInitialLink")) {
+            result.success(initialLink);
+        } else if (call.method.equals("getLatestLink")) {
+            result.success(latestLink);
         } else {
-          events.success(dataString);
+            result.notImplemented();
         }
-      }
-    };
-  }
+    }
+
+    @Override
+    public boolean onNewIntent(Intent intent) {
+        this.handleIntent(context, intent);
+        return false;
+    }
+
+    @Override
+    public void onAttachedToActivity(ActivityPluginBinding activityPluginBinding) {
+        this.handleIntent(this.context, activityPluginBinding.getActivity().getIntent());
```

Comment:
You seem to be missing the new intent listener here and in the reattach
`activityPluginBinding.addOnNewIntentListener(this);`